packages/x-data-grid/src/hooks/features/rowSelection/useGridRowSelection.ts

```diff
@@ -760,11 +826,20 @@ export const useGridRowSelection = (
     // props.isRowSelectable changed
     const currentSelection = gridRowSelectionStateSelector(apiRef);
 
-    if (isRowSelectable) {
-      const newSelection = currentSelection.filter((id) => isRowSelectable(id));
+    if (typeof isRowSelectable === 'function') {
+      let selectableIds = new Set<GridRowId>();
+      if (currentSelection.type === 'include') {
```

Comment:
Should we early-return if `currentSelection.type` is "exclude" as it's not been used?